src/charm.py

```diff
@@ -117,6 +123,55 @@ def _on_config_changed(self, event: ops.HookEvent) -> None:
         """
         self.change_config(event)
         self._set_workload_version()
+        if self._charm_state.enable_mjolnir:
+            self._enable_mjolnir()
```

Comment:
Shouldn't it be smarter ? What if mjolnir was already enabled, you'll systematically call this method which creates a new user, ...
You should only call this method if mjolnir is not already enabled (probably could be deducted from pebble plan service state / scheduling)

Reply:
This definitely makes life easier. I changed it to check if the pebble service is already there.